src/components/form/EncounterFormPanel.jsx

```diff
@@ -230,7 +232,10 @@ EncounterFormPanel.defaultProps = {
 };
 
 const mapStateToProps = (state, props) => {
+  const formValue = getFormValues(props.formInstanceId)(state);
   return {
+    // Check if form has encounter
+    hasEncounter: !!(formValue && formValue['encounter-datetime']),
```

Comment:
The problem with this (I think) is that as soon as you enter an encounter datetime, a "Cancel" will do a "Back"... in "enter" mode (ie, before a form has ever been saved).

What we will want to say is, if the form is in "enter" mode (has not been saved before) a Cancel should Cancel and do a "Back", while if we are in "edit" mode (the form has been previously saved) then a Cancel should clear out any changes during this edit and return to "View" mode.

One way we know if there is an existing encounter is if  the "encounter" prop is passed in, which is was happens in Lab Workflow.

In IC3, this prop is not passed in... but the existing encounter it set up somehow, do you remember how?  (I can't find it).

Reply:
Or if I'm wrong about the above behavior, let me know...

Reply:
> In IC3, this prop is not passed in... but the existing encounter it set up somehow, do you remember how? (I can't find it).

Yes @mogoodrich I can check if the form has an encounter via `this.getForm().encounter`.
I've updated the PR, please take a second look.